lib/request.js

```diff
@@ -57,8 +57,15 @@ var req = exports = module.exports = {
 
 req.get =
 req.header = function header(name) {
-  var lc = name.toLowerCase();
-
+  var lc = null;
+  
+  // If name wasn't a string, it will crash
+  if(typeof name !== 'string') { 
+    new TypeError('Parameter "name" needs to be a String but got a ' + (typeof name));
```

Comment:
You are missing the actual `throw` keyword here, so you actually end up with the same bug you reported :) You'll notice this once you update the tests to test this change, of course :)